source/common/upstream/health_checker_base_impl.cc

```diff
@@ -230,6 +230,7 @@ void HealthCheckerImplBase::ActiveHealthCheckSession::handleSuccess(bool degrade
     // it to healthy. This makes startup faster with a small reduction in overall reliability
     // depending on the HC settings.
     if (first_check_ || ++num_healthy_ == parent_.healthy_threshold_) {
+      host_->healthFlagClear(Host::HealthFlag::PENDING_ACTIVE_HC);
```

Comment:
If someone uses active HC, but has EDS send health state as healthy, what is the initial state? I'm wondering if there is a case where we don't hit this? Would it be better to do this up above like you do in the failure case to make it more clear that this is clearing after the first response?

Reply:
ping on this

Reply:
I think the initial state is still going to be unhealthy (since we don't care about the EDS value when we set the FAILED_ACTIVE_HC flag initially), but it might be clearer and less likely to break in the future if I do what you're suggesting so I'll change it